src/components/map/rent-popup.tsx

```diff
@@ -150,6 +160,19 @@ const RentPopup: React.FC<RentPopupProps> = ({
     [openedStationId],
   );
 
+  useEffect(
+    () => {
+      if (booking && openedStationId && stationDetail && openedStationId === booking.stationId) {
+        const bookedSlot =
+          stationDetail.slots.stationSlots.find(slot => slot.stationSlotPosition === booking.stationSlotPosition);
+        if (bookedSlot) {
+            setSelectedSlot(bookedSlot);
```

Comment:
This can be shortened to `bookedSlot && setSelectedSlot(bookedSlot);`

Reply:
See cf8f15d5f1fdb9cf5bd7e4aed4787aad6ff6d86d.